openpathsampling/engines/external_engine.py

```diff
@@ -48,6 +53,87 @@ def _debug_snapshot_loading(snapshot):
     snapshot.load_details()
     snapshot.clear_cache()
 
+
+class FilenameSetter(StorableNamedObject):
+    """Just use numbers, as we did previously.
+
+    This is the default for compatibility reasons, but it not recommended.
+    Generally, we recommend using :class:`.RandomString`.
+    """
+    # the weird use of this as the base class is because engine options has
+    # some weird type testing that requires replace object to be instances
+    # of the default
+    def __init__(self, count=0):
+        super(FilenameSetter, self).__init__()
+        self.count = count
+
+    def __call__(self):
+        retval = self.count
+        self.count += 1
+        return retval
+
+    def reset(self, count=0):
+        self.count = count
+
+
+class RandomStringFilenames(FilenameSetter):
+    """Use a random string for filename prefixes.
+
+    This is recommended, and will become the default in future versions of
+    OpenPathSampling.
+
+    Parameters
+    ----------
+    length : int
+        number of character in the resulting string
+    allowed : str
+        string containing the allowed characters to return
+    """
+    _allowed = 'abcdefghijklmnopqrstuvwxyz0123456789'
+    def __init__(self, length=8, allowed=None):
+        super(RandomStringFilenames, self).__init__()
+        self.length = length
+        allowed = allowed if allowed is not None else self._allowed
+        self.allowed = np.array([a for a in allowed])
+
+    def __call__(self):
+        return "".join(np.random.choice(self.allowed, self.length))
+
+class _InternalizedEngineProxy(DynamicsEngine):
+    """Wrapper that allows snapshots to be "internalized."
+
+    This is needed because the dynamically-registered snapshot tables are
+    associated to an engine: each engine creates exactly one snapshot table.
+    We use the internalized engine to create those snapshots.
+    """
+    def __init__(self, engine):
+        descriptor = SnapshotDescriptor.construct(
+            snapshot_class=engine.InternalizedSnapshotClass,
+            snapshot_dimensions=engine.descriptor._dimensions
+        )
+        self.engine = engine
+        super(_InternalizedEngineProxy, self).__init__(options={},
+                                                       descriptor=descriptor)
+
+    def to_dict(self):
+        dct = super(_InternalizedEngineProxy, self).to_dict()
+        dct['engine'] = self.engine
+        return dct
+
+    @classmethod
+    def from_dict(cls, dct):
+        # ignore everything except the engine here
+        dct = {'engine': dct['engine']}
+        return super(_InternalizedEngineProxy, cls).from_dict(dct)
```

Comment:
Wouldn't this be easier by just saving the engine? (The you don't have to drop the info again when loading)
```suggestion
    def to_dict(self):
        # We only need the engine to load again
        dct = {'engine': self.engine}
        return dct
```

Reply:
Yeah, I've been trying to figure out if there was some reason I did it this way, but I don't see one.